lib/opencpu/client.rb

```diff
@@ -64,10 +67,21 @@ def process_query(url, data, format, &block)
       case response.code
       when 200..201
         return yield(response)
-      when 400
-        fail '400: Bad Request\n' + response.body
       else
-        fail "#{response.code}:\n #{response.body}"
+        fail error_class_for(response.code), "#{response.code}:\n #{response.body}"
+      end
+    end
+
+    def error_class_for(response_code)
+      case response_code
+      when 403
+        AccessDenied
+      when 400..499
+        BadRequest
+      when 500..599
+        InternalServerError
+      else
+        UnknownError
```

Comment:
Ik zou standaard OpenCPUError terug geven, UnknownError voegt niet heel veel toe, vooral omdat response code toegankelijk is.